inlineplz/linter_runner.py

```diff
@@ -0,0 +1,349 @@
+            if self.should_ignore_path(root):
+                continue
+
+            for filename in filenames:
+                full_path = os.path.join(root, filename)
+                if "text" in identify.tags_from_path(full_path):
+                    paths.add(full_path)
+        return paths
+
+    @staticmethod
+    def should_autorun(config, filenames):
+        patterns = registry.PATTERNS.get(config.get("language"))
+        if config.get("autorun"):
+            for pattern in patterns:
+                if fnmatch.filter(filenames, pattern):
+                    return True
+
+        return False
+
+    def dotfiles_exist(self, config):
+        return any(
+            dotfile.strip() in os.listdir(self.config_dir) for dotfile in config.get("dotfiles")
+        )
+
+    async def install_linter(self, config):
+        install_cmds = config.get("install")
+        for install_cmd in install_cmds:
+            if await self.installed(config):
+                return True
+            if install_cmd in self.previous_install_commands:
+                continue
+
+            self.previous_install_commands.append(install_cmd)
+            try:
+                await self.run_command(install_cmd)
+            except OSError:
+                continue
+
+        return await self.installed(config)
+
+    async def install_trusted(self):
+        for install_cmd in inlineplz.linters.TRUSTED_INSTALL:
+            try:
+                await self.run_command(install_cmd)
+            except OSError:
+                print(
+                    "Trusted install failed: {0}\n{1}".format(install_cmd, traceback.format_exc())
+                )
+                sys.stdout.flush()
+
+    async def installed(self, config):
+        try:
+            returncode, _ = await self.run_command(config.get("help"))
+            return returncode == 0
+        except OSError:
+            return False
+
+    def run_config(self, config):
+        if self.dotfiles_exist(config) and config.get("run"):
+            return config.get("run")
+
+        return [
+            os.path.normpath(item.format(config_dir=self.fallback_config_dir))
+            if "..." not in item
+            else item.format(config_dir=self.fallback_config_dir)
+            for item in (config.get("rundefault") or config.get("run"))
+        ]
+
+    def run_linters(self):  # noqa: MC0001
+        self.cleanup()
+        self.event_loop.run_until_complete(self.performance_hacks())
+
+        if self.trusted and (self.install or self.autorun):
+            self.install_trusted()
+
+        linter_tasks = []
+        for linter in self.linters_to_run():
+            linter_tasks.append(asyncio.ensure_future(self.run_linter(linter), loop=self.event_loop))
+
+        for linter_task in linter_tasks:
+            self.event_loop.run_until_complete(linter_task)
+
+        self.event_loop.close()
+        return self.messages.get_messages()
+
+    async def run_linter(self, linter):
```

Comment:
[inline-plz]: `prospector: mccabe: LinterRunner.run_linter is too complex (12) (MC0001)`